connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java

```diff
@@ -0,0 +1,316 @@
+package com.telenor.mobileconnect;
+
+import android.content.Context;
+import android.net.Uri;
+import android.telephony.TelephonyManager;
+import android.text.TextUtils;
+import android.util.Base64;
+
+import com.squareup.okhttp.HttpUrl;
+import com.telenor.connect.AbstractSdkProfile;
+import com.telenor.connect.id.ConnectAPI;
+import com.telenor.connect.id.ConnectIdService;
+import com.telenor.connect.id.ConnectTokensTO;
+import com.telenor.connect.id.TokenStore;
+import com.telenor.connect.id.UserInfo;
+import com.telenor.connect.utils.ConnectUrlHelper;
+import com.telenor.connect.utils.RestHelper;
+import com.telenor.mobileconnect.id.MobileConnectAPI;
+import com.telenor.mobileconnect.operatordiscovery.OperatorDiscoveryAPI;
+import com.telenor.mobileconnect.operatordiscovery.OperatorDiscoveryConfig;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import retrofit.Callback;
+import retrofit.ResponseCallback;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+public class MobileConnectSdkProfile extends AbstractSdkProfile {
+
+    private OperatorDiscoveryConfig operatorDiscoveryConfig;
+    private volatile OperatorDiscoveryAPI.OperatorDiscoveryResult operatorDiscoveryResult;
+    private volatile OperatorDiscoveryAPI operatorDiscoveryApi;
+
+    public MobileConnectSdkProfile(
+            Context context,
+            final OperatorDiscoveryConfig operatorDiscoveryConfig,
+            boolean useStaging,
+            boolean confidentialClient) {
+        super(context, useStaging, confidentialClient);
+        this.operatorDiscoveryConfig = operatorDiscoveryConfig;
+    }
+
+    @Override
+    public HttpUrl getApiUrl() {
+        String host = operatorDiscoveryResult.getMobileConnectApiUrl().host();
+        if (useStaging) {
+            // will have no effect on non-Telenor subscribers
+            host = host.replace(
```

Comment:
I agree, but I found it handy during my manual tests, since the example app is registered with GSMA for production.